halo2_proofs/src/poly/kzg/commitment.rs

```diff
@@ -132,6 +132,31 @@ where
         }
     }
 
+    /// Initializes parameters for the curve through existing parameters
+    /// k, g, g_lagrange (optional), g2, s_g2
+    pub fn from_parts(
+        &self,
+        k: u32,
+        g: Vec<E::G1Affine>,
+        g_lagrange: Option<Vec<E::G1Affine>>,
+        g2: E::G2Affine,
+        s_g2: E::G2Affine,
+    ) -> Self {
+        let n = 1 << k;
+        Self {
+            k,
+            n,
```

Comment:
No need to declare `n` here. You can just do:
```suggestion
        Self {
            k,
            1 << k,
```